seed/tasks.py

```diff
@@ -864,6 +864,33 @@ def _finish_matching(import_file, progress_key):
     cache.set(progress_key, 100)
 
 
+
+def _normalize_address_str(address_val):
+    """Normalize the address to conform to short abbreviations."""
+    # so the parse function uses a regular expression defined in Regexes to make sure that the address conforms to a standard format, 
+    # the function does not recognize an address unless it has more than just  an address line, so this hack below adds a fake post code, 
+    # a more elegant solution would be to  change the regular expression so that a simple address is ok
+    address_val=address_val +', 95472'
+    #now parse the address into number, street name and street type, and normalize all street types to be
+    addr=sa.parse(address_val)
+    if not addr:
+        return
+    if addr.has_key('number'): 
```

Comment:
Rich,

Im still working on this. Please hold off reviewing it until I comment on
the pull request that im done.

Spencer

On Mon, Mar 23, 2015 at 6:07 PM, Rich Brown <notifications@github.com>
wrote:

> In seed/tasks.py
> <https://github.com/SEED-platform/seed/pull/161#discussion_r26996575>:
>
> > @@ -864,6 +864,33 @@ def _finish_matching(import_file, progress_key):
> >      cache.set(progress_key, 100)
> >
> >
> > +
> > +def _normalize_address_str(address_val):
> > +    """Normalize the address to conform to short abbreviations."""
> > +    # so the parse function uses a regular expression defined in Regexes to make sure that the address conforms to a standard format,
> > +    # the function does not recognize an address unless it has more than just  an address line, so this hack below adds a fake post code,
> > +    # a more elegant solution would be to  change the regular expression so that a simple address is ok
> > +    address_val=address_val +', 95472'
> > +    #now parse the address into number, street name and street type, and normalize all street types to be
> > +    addr=sa.parse(address_val)
> > +    if not addr:
> > +        return
> > +    if addr.has_key('number'):
>
> Will the subsequent append statements fail if the record has no address
> number?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/161/files#r26996575>.
>



-- 
Dr Spencer Dutton
Building 90 R 3058
Indoor Environment
Commercial Ventilation and Indoor Air Quality Group
Lawrence Berkeley National Laboratory
1 Cyclotron Rd, Berkeley, California, USA
Tel:  510 4867179
Email: smdutton@lbl.gov